### DISCUSSION TITLE
Zitadel with OAuth2 Proxy & NGINX-Proxy-Manager = Webpage Unauthorized

### DISCUSSION THREAD

**Discussion by Kryszn0** [zitadel/zitadel#7066] [Q&A]
Hello, hoping for some help to figure out what I'm missing here

I've attached am image below for the general setup at play here. I have managed to get Zitadel working work with OAuth2 Proxy and NGINX-Proxy-Manager, but can't seem to figure out how to have Zitadel authorize access to the original URL I am trying to protect.

Desired steps:
1. Navigate to app.domain.com
2. Be redirected to auth.domain.com to login to Zitadel instance... after successful login...
3. Redirected to app.domain.com

What is actually happening:
1. Navigate to app.domain.com
2. Be redirected to auth.domain.com to login to Zitadel instance... after successful login...
3. Redirected to a URL that generally follows this: auth.domain.com/oauth2/auth/code=${RANDOMSTRING}%https%app.domain.com%
The webpage itself is a black page that simply states 'Unauthorized' on the top left.

Based on the actual outcome in step 3, I believe I have setup the appropriate routing/redirecting as layed out in the image attached. Am I correct to believe that the black webpage with the word 'Unauthorized' in the top left is Zitadel informing me that the user I have logged in as does not have access to app.domain.com? If so, I can't seem to figure out how to tell Zitadel to allow access to app.domain.com .

The Goal:
I'm hoping I can use this setup to have Zitadel used as a front for any of the apps I am self-hosting (AdGuardHome, FileBrowser, Audiobookshelf, etc.). Integrating Zitadel's userbase with the logins of the apps themselves is a later project/task. For now, I just want to have a setup where access to app.domain.com, app2.domain.com, app3.domain.com, etc. is first redirected to Zitadel's login, then after successful login a user with appropriate access to these apps can view/use the webpage. 

[This is the only other thread I can find discussing something similar to my question.
](https://github.com/zitadel/zitadel/discussions/5537)

TIA for any help! 

![proxy](https://github.com/zitadel/zitadel/assets/113187314/1331f367-124e-4e2e-a691-d8a6ab9d5511)

**Comment by hifabienne** [comment]
@eliobischof or @adlerhurst can you help here?

**Comment by adlerhurst** [comment]
Huhu

Cool that your try out zitadel 🥳

It looks like the user is able to sign in.
Did you create a project for app.domain.com? Does it include an app which matches the redirect url? For me it sounds like there is some misconfiguration there. 
Here are some links which should clarify what i mean: 
Project: https://zitadel.com/docs/guides/manage/console/projects
App: https://zitadel.com/docs/guides/manage/console/applications

**Reply by Kryszn0** [reply]
Thanks for taking the time to reply; I was able to figure this out eventually. Required finding a few workarounds for unresolved bugs in NGINX Proxy Manager specifically. I do have a follow-up question now that this is working:

I have my Zitadel instance connected with 3rd party Web applications that are each using an attached OAuth2 Proxy to facilitate login. One thing I've noticed is user fields (e.g., user ID, phone, language, etc.) are not all 'crossing'. I believe Zitadel's user fields are [Objects](https://zitadel.com/docs/apis/actions/objects)? I realize for this crossing to occur, I will need to correctly map any Internal (3rd party application) user fields with the External (Zitadel's) user fields. Are these Objects what I should be mapping? Right now, only First/Last Name & Email are crossing; so I'm thinking I'm just not mapping some fields correctly.

Further, any plans to expand User Fields/Objects? I saw a feature request to include preferred pronouns for HR reasons. Would be good idea imo to add/define additional fields such as: city, country, middle name, mobile phone, second email, institution, department, first name phonetic, last name phonetic, etc.. I'd think the more there are the better it will be in the long-run. Considering Zitadel's intention is to act as an intermediary for other applications/services, user profiles should try to define as many points of identification/description as possible so that profiles across various Projects can be as 1:1 as possible
